### PR TITLE
Fragmentation fixes.

### PR DESCRIPTION
**Pull request by ajamshed**
Signed-off-by: Muhammad Asim Jamshed <muhammad.jamshed@intel.com>

### CODE REVIEW

File: core/modules/ip_frag.cc [hunk 1]
```diff
@@ -120,6 +126,13 @@ IPFrag::FragmentPkt(Context *ctx, bess::Packet *p)
 			for (int i = 0; i < res; i++)
 				EmitPacket(ctx, (bess::Packet *)frag_tbl[i], FORWARD_GATE);
 
+			/* free original mbuf */
+			m = (struct rte_mbuf *)p;
+			do {
+				struct rte_mbuf *del_mbuf = m;
+				m = m->next;
+				rte_pktmbuf_free_seg(del_mbuf);
+			} while (m != NULL);
```
Comment (krsna1729) [inline core/modules/ip_frag.cc:135]: Can we replace with 

```cc
  // With the contexts('ctx'), drop a packet. Dropped packets will be freed.
  inline void DropPacket(Context *ctx, bess::Packet *pkt);
```

https://github.com/NetSys/bess/blob/master/core/module.h#L258-L259

File: core/modules/ip_frag.cc [hunk 2]
```diff
@@ -39,6 +39,12 @@ IPFrag::FragmentPkt(Context *ctx, bess::Packet *p)
 		unsigned char *orig_ip_payload;
 		uint16_t orig_data_offset;
```
Comment (krsna1729) [inline core/modules/ip_frag.cc:41]: Comment on top of the function seems to describe reassembly. Is that right?

Comment (ajamshed) [reply core/modules/ip_frag.cc:43]: Done

File: core/modules/ip_frag.cc (hunk 1 above)
Comment (ajamshed) [reply core/modules/ip_frag.cc]: Done.

**Pull request merged**